src/application/downloader.py

```
@@ -0,0 +1,213 @@
+            max_locations=max_locations
+        )
+        
+        job_id = await self.job_manager.create_job(job)
+        
+        with LogContext(job_id=job_id, country=country_code):
+            try:
+                await self._execute_job(job, resume)
+                await self.job_manager.update_job_status(job_id, "completed")
+                return job_id
+            except Exception as e:
+                logger.error(f"Job {job_id} failed", exc_info=True)
```

Ruff (0.12.2)

Logging `.exception(...)` should be used instead of `.error(..., exc_info=True)`

(G201)